### PR TITLE
fix(resolve-pr-threads): cursor-null typing + totalCount cross-check (#192)

### LINKED ISSUE #192: resolve-pr-threads.sh undercounts unresolved threads → agents miss the conversation-resolution gate
## Problem

`scripts/resolve-pr-threads.sh` (the helper agents use to clear the `required_conversation_resolution: true` gate before merging) silently misses some unresolved review threads. When the script reports "No unresolved threads" but the merge still fails with:
Image
```
gh pr merge 189 --admin --squash --delete-branch
GraphQL: All comments must be resolved. (mergePullRequest)
```

…the agent has no obvious next step, and the human ends up debugging the GraphQL response by hand.

This was reproduced on PR #189:

- `scripts/resolve-pr-threads.sh 189` → "No unresolved threads on PR #189."
- `scripts/resolve-pr-threads.sh 189 --auto-resolve-bots` → same.
- Direct GraphQL `pullRequest.reviewThreads(first: 100)` → returned **3** threads, **1 unresolved** (a `chatgpt-codex-connector` P2 inline finding on `scripts/codex-review-check.sh`).
- The script's GraphQL query (paginating with `first: 50`) returned only **2** threads — it missed the codex P2 thread entirely.

The merge-blocker is invisible from `gh pr checks` (the conversation-resolution gate has no check entry), invisible from `gh pr view --json mergeStateStatus` (just shows `BLOCKED` with no reason text), and now invisible from the script's own list. The only surface that exposes it is the GitHub UI's "Conversations" tab and the `mergePullRequest` GraphQL error itself.

## Hypothesis on the script's miss

Two queries side-by-side, same PR, same auth context, returned different node counts. Possible causes:

1. **Pagination cursor / first: 50 boundary.** The script pages with `first: 50, after: $cursor`; my direct query used `first: 100` no cursor. If the codex thread was on a page boundary or the cursor handling has a bug, the thread would be skipped. Worth confirming with a `first: 100` no-pagination test on the same PR.
2. **`comments(last: 1)` vs `comments(first: 1)` interaction with thread visibility.** The script fetches `comments(last: 1)` for each thread (to anchor the current-HEAD check). If GitHub's GraphQL filters out threads where the last comment is on a deleted/orphaned commit, that could hide threads. Less likely but worth ruling out.
3. **Review-thread vs review-comment distinction.** The codex thread might be a review-level "general comment" rather than an inline thread, but my direct query found it via `reviewThreads.nodes` — so this is unlikely the cause.

The first hypothesis is most likely. Adding a regression that creates 50+ threads on a fixture PR and confirms all are returned would catch this.

## What agents need

Beyond fixing the script, the doctrine needs to give agents a clear escape hatch when the script reports clean but the merge fails. Today's flow:

1. Agent runs `scripts/resolve-pr-threads.sh <PR#>` → "No unresolved threads."
2. Agent attempts merge → "All comments must be resolved."
3. Agent has no documented next step.

The right next step is: **"If the script reports clean but `gh pr merge` errors with `All comments must be resolved`, the script has a blind spot — fall back to a direct GraphQL query for the canonical list."** With a one-liner agents can paste:

```bash
gh api graphql -f query='
query {
  repository(owner: "OWNER", name: "REPO") {
    pullRequest(number: PR_NUM) {
      reviewThreads(first: 100) {
        nodes {
          id
          isResolved
          comments(first: 1) {
            nodes { author { login } path body }
          }
        }
      }
    }
  }
}' --jq '.data.repository.pullRequest.reviewThreads.nodes | map(select(.isResolved == false))'
```

…and then resolve each via `mutation { resolveReviewThread(input: {threadId: "..."}) { thread { isResolved } } }`.

### 1. Fix the pagination bug in `scripts/resolve-pr-threads.sh`

Reproduce the 50-thread boundary case (or whatever the underlying cause is) and patch. Add a unit test that creates a fixture PR with N>50 threads and asserts the script returns all unresolved ones.

### 2. Add a fallback path to the script

When the script returns 0 unresolved but the caller's merge attempt fails with "All comments must be resolved", the script should be re-runnable in a `--paranoid` mode that bypasses its own filtering and pages with `first: 100` (or higher) without the `comments(last: 1)` join — purely listing thread IDs and their `isResolved` flag. If `--paranoid` finds threads the default mode missed, log a warning and resolve them.

Alternative: just always use `first: 100` in the default query — there's no clear reason `first: 50` was chosen, and the GraphQL API supports up to 100 per page.

### 3. Document the fallback in CLAUDE.md § 7.6

Add a note to the existing § 7.6 thread-resolution step that explicitly tells agents what to do when the script reports clean but merge fails:

> If `scripts/resolve-pr-threads.sh` reports "No unresolved threads" AND your merge attempt errors with `All comments must be resolved (mergePullRequest)`, the script has missed a thread. Run the direct GraphQL query in REVIEW_POLICY.md § Manual fallback for unresolved review threads to enumerate, and resolve each via the `resolveReviewThread` mutation.

### 4. Add a section to REVIEW_POLICY.md

`§ Manual fallback for unresolved review threads` — copy-pasteable GraphQL query + mutation, called out as the escape hatch when the script silently undercounts.

### 5. Audit-trail consideration

Per REVIEW_POLICY.md § Implementation notes for branch protection gates, the `--auto-resolve-bots` flag is a clean-up mechanism, not a policy override. The fallback path must preserve that constraint: only bot-authored threads where the finding has been addressed on the current HEAD may be resolved. Human-authored threads still require UI-only resolution. The fallback query above returns the author so agents can apply the same gate.

## Acceptance

- [ ] `scripts/resolve-pr-threads.sh` returns the same set of unresolved threads as a direct `reviewThreads(first: 100)` GraphQL query on PR #189 (and any other fixture PR with comparable thread topology).
- [ ] Unit test in `scripts/ci/check_resolve_pr_threads` (new or existing) covers the fixture case that exposed the bug.
- [ ] CLAUDE.md § 7.6 includes the fallback escape-hatch instruction.
- [ ] REVIEW_POLICY.md has a `§ Manual fallback for unresolved review threads` subsection with copy-pasteable queries.
- [ ] On the next PR where conversation resolution becomes the merge blocker, the agent can self-recover without human-debug intervention.

## References

- #166 — original issue that introduced `scripts/resolve-pr-threads.sh`
- #181, #190, #192 — observed cases of the conversation-resolution blocker (per the script's own header)
- PR #189 — the case that exposed the script's blind spot (codex P2 thread on `scripts/codex-review-check.sh`)
- #191 — auto-clear blocking labels (sibling automation work; both reduce manual-merge friction)

## Immediate workaround (until this lands)

Agents who hit `mergePullRequest: All comments must be resolved` after the script reports clean should paste:

```bash
gh api graphql -f query='
query {
  repository(owner: "nathanjohnpayne", name: "mergepath") {
    pullRequest(number: <PR#>) {
      reviewThreads(first: 100) {
        nodes { id isResolved comments(first: 1) { nodes { author { login } path body } } }
      }
    }
  }
}' --jq '.data.repository.pullRequest.reviewThreads.nodes | map(select(.isResolved == false))'
```

Then for each unresolved bot-authored thread where the finding has been addressed on the current HEAD:

```bash
gh api graphql -f query='
mutation { resolveReviewThread(input: {threadId: "<THREAD_ID>"}) { thread { isResolved } } }'
```

### PR DESCRIPTION
**Pull request by nathanjohnpayne**
Closes #192.

## Root cause

Misdiagnosed in the issue body as eventual consistency. Real cause: an argument-typing bug.

The script paginated `reviewThreads` with `-f cursor="$CURSOR"`. On the first call `$CURSOR="null"` (the literal string). `gh api`'s `-f` flag sends values as strings, so GitHub received `cursor: "null"` (JSON string), interpreted it as a real opaque cursor, and silently returned the wrong thread set. Reproduced deterministically on closed PR #189: `totalCount=3` but only 2 nodes returned by the script.

Direct verification:
```
gh api graphql ... -f cursor=null  → returns 2 nodes (BUG)
gh api graphql ... -F cursor=null  → returns 3 nodes (CORRECT)
```

## Fix

- **First call uses `-F cursor=null`** (typed conversion → real GraphQL null). Subsequent pages use `-f cursor="$CURSOR"` (real cursor string from the prior `endCursor`).
- **`totalCount` cross-validation**: if `returned < totalCount`, exit 2 with a clear undercount error instead of the silent "no unresolved threads" output that masked the bug. Belt-and-suspenders against future API quirks.
- **`comments(last: 1)` → `comments(first: 50)`**: threads almost never exceed 50 comments; one fetch gives both first comment (original review for excerpt) and last comment (HEAD anchor for current-HEAD check).
- **Defensive `select(.isResolved != true)`**: null/missing field treated as unresolved (prefer noise over silent skip).

## Test coverage

New `scripts/ci/check_resolve_pr_threads` with 5 tests:
1. First-call argv contains `-F cursor=null` (positive)
2. First-call argv does NOT contain `-f cursor=null` (negative — closes the regression)
3. `totalCount > returned` → exit 2 with undercount error
4. `totalCount == returned` (all resolved) → exit 0 + clean message
5. `isResolved: null` → treated as unresolved (defensive)

Wired into `repo_lint.yml`. While there: also wired the previously-unrun `check_phase_4b_classifier` (committed in PR #190 but no workflow ever invoked it — separate gap).

## Doc

`CLAUDE.md § 7.6` now includes the manual GraphQL escape hatch agents can paste when the script reports clean but the merge errors with `All comments must be resolved`. Explicit fallback for the case where this script's defenses ever get bypassed.

## Self-Review

- **Correctness**: cursor-null fix verified by direct API test on PR #189 data. `totalCount` cross-validation reproduces the original bug condition under test.
- **Regression risk**: low — existing exit codes and CLI surface unchanged; new paths are additive.
- **Coverage**: 5 new tests; wired into CI alongside two previously-unwired test suites.
- **Style**: matches existing `scripts/ci/check_*` shape (PATH-prepended gh stub, set +e/-e brackets).
- **Security/dependency hygiene**: no new deps; no auth changes.

Authoring-Agent: claude

### CODE REVIEW

File: scripts/resolve-pr-threads.sh [hunk 1]
```diff
@@ -139,62 +139,126 @@ HEAD_OID=$(gh api "repos/$OWNER/$NAME/pulls/$PR_NUM" --jq .head.sha 2>/dev/null)
...
+    repository(owner: $owner, name: $repo) {
+      pullRequest(number: $pr) {
+        reviewThreads(first: 50, after: $cursor) {
+          totalCount
+          pageInfo { hasNextPage endCursor }
+          nodes {
+            id
+            isResolved
+            isOutdated
+            comments(first: 50) {
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/resolve-pr-threads.sh]: Addressed in 3e5674b. Split the GraphQL into dual aliases:

- `commentsFirst: comments(first: 1)` — original review for excerpt + author display.
- `commentsLast: comments(last: 1)` — guaranteed-latest commit_oid for HEAD anchor.

Deterministic for any thread depth (no >50-comments edge case). Updated test stubs to the new alias shape and added a 6th regression test (#5 in the suite, ordered last) exercising the scenario you flagged. 6/6 tests pass.

File: CLAUDE.md [hunk 2]
```diff
@@ -141,6 +141,44 @@ keyring active is your agent identity. No switch needed for commits.
...
+                 comments(first: 1) {
+                   nodes { author { login } path body }
+                 }
+               }
+             }
+           }
+         }
+       }' --jq '.data.repository.pullRequest.reviewThreads.nodes
+                 | map(select(.isResolved != true))'
+     ```
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply CLAUDE.md:181]: Addressed in 725e497. Added `totalCount` + `pageInfo { hasNextPage endCursor }` to the escape-hatch query plus a jq guard that fails loudly if `hasNextPage` is true:

```jq
if .data.repository.pullRequest.reviewThreads.pageInfo.hasNextPage
then "ERROR: PR has >100 review threads; paginate with after:\$endCursor"
     | halt_error(2)
else .data.repository.pullRequest.reviewThreads.nodes
     | map(select(.isResolved != true))
end
```

For the very rare >100-thread case the operator must paginate, but the common-case copy-paste stays one query.

**Comment by nathanpayne-claude** [comment]
@codex review

File: CLAUDE.md [hunk 3]
```diff
@@ -141,6 +141,55 @@ keyring active is your agent identity. No switch needed for commits.
...
+           }
+         }
+       }' --jq '
+         if .data.repository.pullRequest.reviewThreads.pageInfo.hasNextPage
+         then "ERROR: PR has >100 review threads; paginate with after:$endCursor"
+              | halt_error(2)
+         else .data.repository.pullRequest.reviewThreads.nodes
+              | map(select(.isResolved != true))
+         end
+       '
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply CLAUDE.md:180]: Addressed in adb5423 — prefixed the escape-hatch query with `GH_TOKEN="\$OP_PREFLIGHT_REVIEWER_PAT"` so operators consistently use the preflight reviewer PAT, not the active keyring account.

File: scripts/resolve-pr-threads.sh [hunk 4]
```diff
@@ -139,62 +139,139 @@ HEAD_OID=$(gh api "repos/$OWNER/$NAME/pulls/$PR_NUM" --jq .head.sha 2>/dev/null)
...
+while :; do
+  if [ -z "$CURSOR" ]; then
+    PAGE=$(gh api graphql -f query="$QUERY" \
+      -F owner="$OWNER" -F repo="$NAME" -F pr="$PR_NUM" -F cursor=null 2>&1) || {
+      echo "GraphQL query failed: $PAGE" >&2
+      exit 2
+    }
+  else
+    PAGE=$(gh api graphql -f query="$QUERY" \
+      -F owner="$OWNER" -F repo="$NAME" -F pr="$PR_NUM" -f cursor="$CURSOR" 2>&1) || {
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/resolve-pr-threads.sh:247]: Addressed in adb5423 — both pagination branches now prefix `gh api graphql` with `GH_TOKEN="\${OP_PREFLIGHT_REVIEWER_PAT:-\${GH_TOKEN:-}}"`. Falls through to the active keyring only when preflight wasn't run (env var unset/empty).

File: scripts/resolve-pr-threads.sh [hunk 5]
```diff
@@ -139,62 +139,139 @@ HEAD_OID=$(gh api "repos/$OWNER/$NAME/pulls/$PR_NUM" --jq .head.sha 2>/dev/null)
...
+  cat >&2 <<EOF
+ERROR: GraphQL undercount detected on $REPO#$PR_NUM.
+       reviewThreads.totalCount = $TOTAL_COUNT, but the paginated query
+       returned $RETURNED_COUNT nodes. This is the eventual-consistency /
+       cache-shape failure mode that masked unresolved threads on
+       PR #189 (May 2026 — see scripts/resolve-pr-threads.sh header).
+       Do NOT trust the "no unresolved threads" output below; fall back
+       to the manual GraphQL escape hatch in CLAUDE.md § 7.6.
+EOF
+  exit 2
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/resolve-pr-threads.sh]: Addressed in adb5423 — switched the cross-validation to equality (`RETURNED_COUNT != TOTAL_COUNT`) so over-count (duplicate page / cursor reset) fails too. Updated existing test to match the new error message and added test 4b that exercises the over-count case with a stub returning 2 nodes when totalCount=1. 7/7 tests pass.

**Comment by nathanpayne-claude** [comment]
## External Review Required

**PR:** #194 — fix(resolve-pr-threads): cursor-null typing + totalCount cross-check (#192)
**Branch:** `fix/resolve-pr-threads-blind-spot-192`
**Author Agent:** claude

### Summary

Closes #192 (resolve-pr-threads.sh undercounts unresolved threads). PR shipped through 3 author-side review rounds:

- **r1** (c3996dc): cursor-null typing fix + totalCount cross-validation + commentsFirst/commentsLast alias rationale + 5-test regression suite.
- **r2** (3e5674b): codex P2 — split GraphQL into dual aliases (`commentsFirst: comments(first: 1)` + `commentsLast: comments(last: 1)`) so HEAD-anchor check is deterministic for any thread depth, not just ≤50 comments. Added 6th regression test.
- **r3** (725e497 + adb5423): CR Major × 2 + Minor — pinned both pagination GraphQL reads to `OP_PREFLIGHT_REVIEWER_PAT`, switched totalCount cross-check from `<` to `!=` (catches over-count from duplicate-page regressions too), prefixed CLAUDE.md escape-hatch with same PAT pattern. Added 7th regression test for over-count.

7/7 tests pass on the test suite. Wired both `check_resolve_pr_threads` and the previously-unwired `check_phase_4b_classifier` into `repo_lint.yml`.

### Focus areas

1. **Cursor-null root cause** ([scripts/resolve-pr-threads.sh:142-167](https://github.com/nathanjohnpayne/mergepath/blob/fix/resolve-pr-threads-blind-spot-192/scripts/resolve-pr-threads.sh)). The `-F cursor=null` (typed) vs `-f cursor=null` (string) distinction was the actual root cause of the PR #189 silent undercount — initially misdiagnosed in the issue body as eventual consistency. Worth tracing the cursor-state branching logic and confirming the fallback to active keyring (when preflight wasn't run) is the right behavior.

2. **totalCount equality check vs `<` check** ([scripts/resolve-pr-threads.sh:244-256](https://github.com/nathanjohnpayne/mergepath/blob/fix/resolve-pr-threads-blind-spot-192/scripts/resolve-pr-threads.sh)). CR Major flagged that `<` only catches under-count; over-count from duplicate-page regression slips through. Switched to `!=`. Worth confirming this can't false-fire on legitimate transitions (e.g., a thread being created mid-pagination).

3. **PAT pinning fallback semantics** ([scripts/resolve-pr-threads.sh:223-231](https://github.com/nathanjohnpayne/mergepath/blob/fix/resolve-pr-threads-blind-spot-192/scripts/resolve-pr-threads.sh)). Both pagination branches now do `GH_TOKEN="${OP_PREFLIGHT_REVIEWER_PAT:-${GH_TOKEN:-}}"`. The fallback chain: preflight reviewer PAT → caller's GH_TOKEN → active keyring. Worth confirming that's the right precedence vs. just always preferring preflight when available.

### Observations from internal review

- All 7 unit tests pass including the 3 new regressions added during the r2/r3 cycles.
- The new `check_resolve_pr_threads` is wired into `repo_lint.yml`, alongside `check_phase_4b_classifier` (which had been committed in PR #190 but never wired — incidental bonus fix).
- The CLAUDE.md § 7.6 escape hatch now has both `totalCount` + `pageInfo.hasNextPage` guards AND the GH_TOKEN pinning, mirroring the script's defenses for the manual-fallback case.
- Codex App didn't return on either r1 or r2 — same flakiness pattern as on PRs #171, #173, #179, #180, #181, #189, #190.

### Suggested external reviewer

nathanpayne-codex

### Rationale for external review

Phase 4a Codex App posted r1 P2 (addressed in 3e5674b) but didn't return on r2 retrigger after 598s. Same flakiness pattern as on prior PRs in this session. Phase 4b for an independent CLI verdict on the r2 + r3 hardening fixes.

— Authoring agent: claude. Human relays to codex CLI session for review.

**Review by nathanpayne-cursor — DISMISSED** [review]
Approved from Cursor review.

I reviewed the current head (`adb5423`) across `scripts/resolve-pr-threads.sh`, `scripts/ci/check_resolve_pr_threads`, `repo_lint.yml`, and the `CLAUDE.md` escape-hatch docs. I do not see a blocking correctness issue in the cursor-null fix, count cross-check, comment aliasing, or CI wiring.

Verification performed locally:
- `./scripts/ci/check_resolve_pr_threads`
- `./scripts/ci/check_phase_4b_classifier`
- `./scripts/ci/check_workflow_parsers`

Notes on remaining bot threads:
- The two-page pagination fixture suggestion would strengthen coverage, but I do not consider it blocking for this PR because the implementation still paginates, validates total count equality, and keeps the second-page cursor branch straightforward.
- I verified `GH_TOKEN="" gh api user` falls back to the active keyring account on this machine, so the empty-token concern is not reproducing as a behavior bug here.
- The unpinned manual `resolveReviewThread` mutation is consistent with this repo's active-account convention for write paths; the read-path query is the part that needs explicit PAT pinning.

File: CLAUDE.md [hunk 6]
```diff
@@ -141,6 +141,56 @@ keyring active is your agent identity. No switch needed for commits.
...
+     ```
+
+     For each unresolved bot-authored thread where the finding is
+     addressed on the current HEAD, resolve via:
+
+     ```bash
+     gh api graphql -f query='mutation { resolveReviewThread(input:
+       {threadId: "THREAD_ID"}) { thread { isResolved } } }'
+     ```
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply CLAUDE.md:191]: Addressed in e6768c8 — pinned the escape-hatch mutation to `GH_TOKEN="\$OP_PREFLIGHT_REVIEWER_PAT"` matching the read query above. Mutation is reviewer-attributed.

File: scripts/ci/check_resolve_pr_threads [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+fi
+
+echo
+if [ "$fail" -eq 0 ]; then
+  echo "check_resolve_pr_threads: PASS ($pass tests)"
+  exit 0
+else
+  echo "check_resolve_pr_threads: FAIL ($fail of $((pass + fail)) tests)" >&2
+  exit 1
+fi
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/ci/check_resolve_pr_threads:455]: Addressed in e6768c8 — added test 6 (two-page pagination). Stub returns page 1 with `hasNextPage:true` + `endCursor:CURSOR1`, page 2 with `hasNextPage:false`. Asserts (a) script exits 0 with accumulated count matching totalCount AND (b) the 2nd call's argv contains `-f cursor=CURSOR1`. Catches paginator-on-happy-path regressions.

File: scripts/resolve-pr-threads.sh [hunk 8]
```diff
@@ -139,62 +139,148 @@ HEAD_OID=$(gh api "repos/$OWNER/$NAME/pulls/$PR_NUM" --jq .head.sha 2>/dev/null)
...
+      -F owner="$OWNER" -F repo="$NAME" -F pr="$PR_NUM" -F cursor=null 2>&1) || {
+      echo "GraphQL query failed: $PAGE" >&2
+      exit 2
+    }
+  else
+    PAGE=$(GH_TOKEN="${OP_PREFLIGHT_REVIEWER_PAT:-${GH_TOKEN:-}}" \
+      gh api graphql -f query="$QUERY" \
+      -F owner="$OWNER" -F repo="$NAME" -F pr="$PR_NUM" -f cursor="$CURSOR" 2>&1) || {
+      echo "GraphQL query failed: $PAGE" >&2
+      exit 2
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/resolve-pr-threads.sh]: Good catch — addressed in e6768c8. Replaced the `GH_TOKEN="\${VAR:-\${OTHER:-}}"` pattern with a `gh_read` wrapper that conditionally sets GH_TOKEN only when a non-empty token exists. Also added test 7 that explicitly unsets both PAT vars in the subshell and asserts gh sees GH_TOKEN unset (not empty) — directly validates the keyring-fallback path.

File: scripts/resolve-pr-threads.sh [hunk 9]
```diff
@@ -139,62 +139,158 @@ HEAD_OID=$(gh api "repos/$OWNER/$NAME/pulls/$PR_NUM" --jq .head.sha 2>/dev/null)
...
+      -F owner="$OWNER" -F repo="$NAME" -F pr="$PR_NUM" -F cursor=null 2>&1) || {
+      echo "GraphQL query failed: $PAGE" >&2
+      exit 2
+    }
+  else
+    PAGE=$(gh_read api graphql -f query="$QUERY" \
+      -F owner="$OWNER" -F repo="$NAME" -F pr="$PR_NUM" -f cursor="$CURSOR" 2>&1) || {
+      echo "GraphQL query failed: $PAGE" >&2
+      exit 2
+    }
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/resolve-pr-threads.sh]: Addressed in 39e70a7 — hoisted `READ_GH_TOKEN` + `gh_read()` above the first gh call (formerly line 114 `gh repo view`). Both earlier calls now use `gh_read`. Removed the duplicate definition that was inline above the pagination loop. Centralized so any future read-path gh call gets the same auth treatment by default. 9/9 tests still pass.

**Review by nathanpayne-cursor — APPROVED** [review]
Re-approved as nathanpayne-cursor on HEAD 39e70a7e820fe3cf57475f7dd17df2b2b67b05ff.

Reviewed the delta since my prior approval: the manual escape-hatch mutation is now explicitly pinned to the reviewer PAT, read-path `gh` calls use the centralized `gh_read` wrapper without the empty-token fallback trap, and the regression suite now covers two-page pagination plus keyring fallback behavior.

Local verification on this head:
- `./scripts/ci/check_resolve_pr_threads`
- `./scripts/ci/check_phase_4b_classifier`
- `./scripts/ci/check_workflow_parsers`
